File: google_script/create_ticket.js
function testCreateTicket() {
  
  var API_KEY = 'YOUR_API_KEY';
  var ENDPOINT = 'http://YOUR_DOMAIN.freshdesk.com'; // change to https if SSL is configured.
  
  var headers = {
    'Content-type': 'application/json',
    'Authorization': 'Basic ' + Utilities.base64Encode(API_KEY + ':X')
  };
  
  var payload = '{"helpdesk_ticket":{"description":"Description ...","subject":"Title","email":"example@example.com","priority":1,"status":2},"cc_emails":"example1@example.com"}';
  var url = ENDPOINT + '/helpdesk/tickets.json';
  
  var options = {
    'method': 'post',
    'headers': headers,
    'payload': payload,
    muteHttpExceptions: true
  };
  
  var response = UrlFetchApp.fetch(url, options);
  
  Logger.log("code: " + response.getResponseCode());
  Logger.log("text: " + response.getContentText());
}

File: python_samples/create_ticket.py
## This script requires "requests": http://docs.python-requests.org/
## To install: pip install requests

import json
import requests

FRESHDESK_ENDPOINT = "http://YOUR_DOMAIN.freshdesk.com" # check if you have configured https, modify accordingly
FRESHDESK_KEY = "YOUR_API_KEY"

headers = {'Content-Type': 'application/json'}

payload = {
    'helpdesk_ticket': {
        'subject': 'Ticket title',
        'description': 'Ticket detail',
        'email': 'example@example.com',
        'priority': 1,
        'status': 2
    },
    'cc_emails': 'example1@example.com'
}

print json.dumps(payload)

r = requests.post(FRESHDESK_ENDPOINT + '/helpdesk/tickets.json',
        auth=(FRESHDESK_KEY, "X"),
        headers=headers,
        data=json.dumps(payload))

print 'HTTP response code: ' + str(r.status_code)
o = json.loads(r.content)
print 'New ticket id: ' + str(o['helpdesk_ticket']['display_id'])
